### PR TITLE
feat: Display ticket price range and revalidate tickets

### LINKED ISSUE #1990: Display ticket prices in event details fragment
To match with Eventbrite

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #1990

<p align="center">
 
</p>

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 1]
```diff
@@ -84,6 +86,9 @@ class EventsFragment : Fragment(), BottomIconDoubleClick {
             .nonNull()
             .observe(this, Observer { list ->
                 eventsListAdapter.submitList(list)
+                list.forEach {
+                    Timber.e(it.ticketPrices.toString())
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: I'm getting app prices to 0.0 - 
```
[TicketPrice(id=820, price=0.0), TicketPrice(id=821, price=0.0), TicketPrice(id=822, price=0.0), TicketPrice(id=823, price=0.0), TicketPrice(id=824, price=0.0), TicketPrice(id=830, price=0.0), TicketPrice(id=819, price=0.0)]
```

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: On chrome dev tools - 
image

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: @iamareebjamal Is relationship work like that? It is giving correct if but the price to zero.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: You have just requested id so why will you get prices.

Besides, do you realize how bad this solution is? You'll be loading all tickets for every event this way

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: > You'll be loading all tickets for every event this way

But it is loading id and price only for tickets which are required in event details fragment. And almost all variables in event details fragment are loading from the database.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: > You have just requested id so why will you get prices.

How I have added `?include=tickets`

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Give me the event ID

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: > Give me the event ID

It is for all events. I have just pasted one.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: https://api.eventyay.com/v1/events/833?include=tickets

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Can I do one thing, Just fetch all tickets of events, insert them into the database and use same in tickets fragment? Also, revalidate tickets in the ticket fragment.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Yes, you can do that but only in event detail fragment

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Also, separate the event detail query from ticket fetching query as it'll take a long time to query event with all the tickets

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Updated

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt [hunk 2]
```diff
@@ -255,6 +261,50 @@ class EventDetailsViewModel(
...
+            .withDefaultSchedulers()
+            .subscribe({
+                loadPriceRange(it, event.paymentCurrency.toString())
+            }, {
+                Timber.e(it, "Error fetching tickets")
+            })
+    }
+
+    private fun loadPriceRange(tickets: List<Ticket>, paymentCurrency: String) {
+        var minPrice = tickets[0].price
```
Comment (anhanh11001) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt]: What happens if an event doesn't have ticket yet?

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 3]
```diff
@@ -212,6 +216,13 @@ class EventDetailsFragment : Fragment() {
                 setHasOptionsMenu(true)
             })
 
+        eventViewModel.priceRange
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                rootView.ticketPriceLinearLayout.visibility = View.VISIBLE
```
Comment (anhanh11001) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Use isVisible

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt (hunk 2 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt]: Will return if the list is empty :+1:

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt]: Updated

File: app/src/main/res/layout/content_event.xml [hunk 4]
```diff
@@ -184,6 +184,34 @@
...
             </LinearLayout>
 
+            <LinearLayout
+                android:id="@+id/ticketPriceLinearLayout"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/layout_margin_large"
+                android:layout_marginEnd="@dimen/layout_margin_medium"
+                android:visibility="gone"
+                android:background="?selectableItemBackground"
```
Comment (anhanh11001) [inline app/src/main/res/layout/content_event.xml]: I think this can be removed as clicking on it doesn't do anything right

**Comment by liveHarshit** [comment]
> Write DB query to find min and max price

Updated

**Pull request merged**